parlai/agents/vsepp_caption/vsepp_caption.py

```diff
@@ -150,58 +150,22 @@ def candidate_helper(self, candidate_vecs, candidate_labels, is_testing):
 
         return (padded_cands, cands, cand_lens, truth_idx)
 
-    def batch_act(self, observations):
-        batch_size = len(observations)
-        # initialize a table of replies with this agent's id
-        batch_reply = [{'id': self.getID()} for _ in range(batch_size)]
-
-        if any(['labels' in obs for obs in observations]):
-            self.mode = 'train'
-        elif any(['eval_labels' in obs for obs in observations]):
-            self.mode = 'valid'
-        else:
-            self.mode = 'test'
-
-        vec_obs = [self.vectorize(obs)
-                   for obs in observations]
-
-        # shift the labels into the text field so they're ordered
-        # by length
-        for item in observations:
-            if self.mode == 'train':
-                if 'labels' in item:
-                    item['text'] = item['labels'][0]
-                    item['text_vec'] = item['labels_vec']
-            elif self.mode == 'valid':
-                if 'eval_labels' in item:
-                    item['text'] = item['eval_labels'][0]
-                    item['text_vec'] = item['eval_labels_vec']
-
-        batch = self.batchify(vec_obs, sort=True)
-
-        images = torch.stack([self.transform(observations[idx]['image'])
-                              for idx in batch.valid_indices])
+    def batchify(self, *args, **kwargs):
+        kwargs['sort'] = True
+        return super().batchify(*args, **kwargs)
+
+    def train_step(self, batch):
+        images = torch.stack([self.transform(img) for img in batch.image])
         if self.use_cuda:
             images = images.cuda(async=True)
```

Comment:
can you confirm async works still? I remember removing it after a torch version update in the past, but I'm not sure

Reply:
confirmed it works!